### ISSUE TITLE
Angular2 Ngc error while importing DragulaModule

### ISSUE THREAD

**Issue by mayconjcm** [valor-software/ng2-dragula#443]
Hello,

I'm using Dragula (module and service) with platformBrowserDynamic for Development, and is working very very fine!!

But when i try to **Build** my App, using prod platformBrowser, the **ngc** shows me this error: 
`ngc: Error: Unexpected value 'DragulaModule' imported by the module 'AppModule'`

Someone also got this error?

Thanks, everyone!

**Comment by audan2009** [comment]
Yes! I was just submitting a issue to Ionic... not sure if the issue is with Dragula or Ionic...

**Comment by audan2009** [comment]
Here is a repo

https://github.com/audan2009/ionic2-dragula

**Comment by mayconjcm** [comment]
I'm getting this error with And without ionic. The problem is The Dragula:(

**Comment by audan2009** [comment]
Ah ok I'll close the Ionic one then

**Comment by prabhap** [comment]
Yes. I am also getting the same error when I tried to run angular2 AoT. Please help

**Comment by mayconjcm** [comment]
Damm :(

**Comment by audan2009** [comment]
I can't try this until later today but I wonder if this would work:

http://blog.angular-university.io/angular2-ngmodule/

Change the ng2-dragula.ts file from:

import { NgModule } from '@angular/core';
import { DragulaDirective } from './components/dragula.directive';
import { DragulaService } from './components/dragula.provider';

export * from './components/dragula.provider';
export * from './components/dragula.directive';

@NgModule({
  exports: [DragulaDirective],
  declarations: [DragulaDirective],
  providers: [DragulaService]
})
export class DragulaModule {
}

to


import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { DragulaDirective } from './components/dragula.directive';
import { DragulaService } from './components/dragula.provider';

export * from './components/dragula.provider';
export * from './components/dragula.directive';

@NgModule({
  exports: [DragulaDirective],
  declarations: [DragulaDirective],
  imports: [BrowserModule],
  providers: [DragulaService]
})
export class DragulaModule {
}

**Comment by mayconjcm** [comment]
I will try this in some minutes...

**Comment by audan2009** [comment]
ya that doesn't work

**Comment by audan2009** [comment]
this looks useful

https://medium.com/@isaacplmann/getting-your-angular-2-library-ready-for-aot-90d1347bcad#.rhhj3bo2d

**Comment by mayconjcm** [comment]
Nice article, @audan2009 . I think we need to fix the dragulaModule to work with AoT.
I'm analysing the module to discover where to repair And make a Pull Request.

My app is still broken because of this problem, so it will have high priority here.

**Comment by audan2009** [comment]
Way above my head to actually implement. Thanks for helping!

**Comment by dany-sh** [comment]
Any news about this issue ?

**Comment by tamascsaba** [comment]
@NathanWalker @valorkin ??

**Comment by valorkin** [comment]
Guys, if something is urgent, mention me please...
I have 300+ github notifications per day, it is hard to follow
I will add metadata today

**Comment by tamascsaba** [comment]
sorry :( but it was no answer from member.

**Comment by valorkin** [comment]
np, working on it

**Comment by valorkin** [comment]
published as 1.2.2-0
beta
bundles build failed, I will fix it later
please check that AoT works fine

**Comment by mayconjcm** [comment]
Here is working!

Thank you, @valorkin !

**Comment by mayconjcm** [comment]
@audan2009 and @valorkin , can i close this issue? Its fine the AOT!

**Comment by audan2009** [comment]
Yes that sounds great! Haven't had a chance to test ionic but it'll probably work. I can test it tomorrow.

**Comment by vdclouis** [comment]
@valorkin any chance you could make a npm release?

**Comment by audan2009** [comment]
@mayconjcm it works!

Thanks for the help @valorkin !!

**Comment by kgiroux** [comment]
Any news about a NPM release ? 

Thank in advance
